Yoast/wordpress-seo
# Implement ARIA tabbed interface for the new metabox tabs

## Summary

This PR can be summarized in the following changelog entry:

* Implement ARIA tabbed user interface for the new meta box tabs.

## Relevant technical choices:
- the JavaScript part that handles the tabs is actually in jQuery, this creates some limitations
- I did my best to integrate with the existing code and at the same time try to make the new code reusable or, at least, easy to be made reusable in future iterations
- the tabs now have proper semantics and the expected keyboard interaction
- for a detailed explanation of the ARIA tabbed interface pattern, see the ARIA authoring practices: https://www.w3.org/TR/wai-aria-practices-1.1/#tabpanel
- for a simpler explanation, see the image below:

![tabbed-interface-from-inclusive-components-design](https://user-images.githubusercontent.com/1682452/59422975-81fc7d00-8dd1-11e9-996d-6a26cec12f36.png)

[image from https://inclusive-components.design/tabbed-interfaces/]

- the tabs have _only one_ tab stop
- left and right arrows loop through the tabs and activate them
- Home key (fn + Left arrow on a mac): jump to the first tab
- End key (fn + Right arrow on a mac): jump to the last tab
- Spacebar: activates the tab if it was not activated automatically on focus: this is an edge case for screen readers, as screen readers can move to elements without actually moving focus
- note: also vertical tabs are supported (they work with Up and Down arrow keys)

Worth noting we already use ARIA tabs in the Help Center (implemented with react-tabs) with basically the same pattern.

Design note:

Ideally, when pressing tab to move to the tab content (the so called "tab panel"), the panel should have a focus style. To clarify, this is [the example from the W3C](https://www.w3.org/TR/wai-aria-practices-1.1/examples/tabs/tabs-1/tabs.html) (see the orange outline):

<img width="435" alt="Screenshot 2019-06-13 at 11 32 57" src="https://user-images.githubusercontent.com/1682452/59424836-000e5300-8dd5-11e9-8124-4d2ac14ac04f.png">

As this part requires design feedback, I'll split it in a separate issue.

## Test instructions
- activate the News and Video add-ons
- test the tabs both in the post, media, and taxonomy meta boxes
- first: verify clicking the tabs works as before
- then, using the keyboard:
- tab to the tabs 😄 
- use left and arrow keys to navigate sequentially through the tabs and activate them
- when on the last or first tab, verify that keeping pressing the arrow keys loops through the tabs
- verify Home and End keys jump to first and last tabs

Then, inspect the tabs in the dev tools
- verify that on page load the inactive tabs have `aria-selected="false" tabindex="-1"`
- verify that on page load the active tab has only `aria-selected="true"`
- verify that when switching tabs, the active one gets `aria-selected="true"` and no tabindex, while the other ones get `aria-selected="false" tabindex="-1"`
- verify all the tabs have an `aria-controls` attribute pointing to the correct tab panel, for example `aria-controls="wpseo-meta-section-content"`
- verify the tab panels have an `aria-labelledby` attribute pointing to the correct tab, for example `aria-labelledby="wpseo-meta-tab-content"`

Then, using Safari + VoiceOver (or other good comoos):
- verify the tabs number and position is announced: 1 of 4, 2 of 4...
- verify the active tab is announced as "selected":

<img width="634" alt="Screenshot 2019-06-13 at 12 10 25" src="https://user-images.githubusercontent.com/1682452/59424564-7cecfd00-8dd4-11e9-9fb1-f9b37160bfaf.png">


- once on a tab, press the Tab key to move into the associated tab panel and verify it's announced as "tab panel" with the correct name (e.g. "Social"):

<img width="721" alt="Screenshot 2019-06-13 at 12 11 20" src="https://user-images.githubusercontent.com/1682452/59424662-af96f580-8dd4-11e9-9db7-ea519036cc9e.png">


## UI changes


Fixes #13106
